src/main/java/org/opentripplanner/routing/edgetype/TimetableResolver.java

```diff
@@ -50,6 +52,47 @@ public int compare(Timetable t1, Timetable t2) {
             return t1.serviceDate.compareTo(t2.serviceDate);
         }
     }
+    
+    /**
+     * Class to use as key in HashMap containing trip id and service date
+     */
+    protected class TripIdAndServiceDate {
```

Comment:
Simple hashtable key classes like this are a perfect example of why people mock Java for being verbose. Of course, it is often more readable to have a special-purpose named class as the hashtable key. When the name of the class is just the concatenation of its constituent types, consider using T2<TypeA, TypeB>. It should also be possible to do `TripIdAndServiceId extends T2<String, ServiceDate>` to reuse the T2 (2-tuple) code.